packages/lib-classifier/src/store/FeedbackStore.js

```diff
@@ -1,45 +1,76 @@
 import { autorun } from 'mobx'
-import { addDisposer, getRoot, onAction, types } from 'mobx-state-tree'
+import { addDisposer, addMiddleware, getRoot, onAction, types } from 'mobx-state-tree'
+import { flatten } from 'lodash'
 
 import helpers from './feedback/helpers'
 import strategies from './feedback/strategies'
 
 const FeedbackStore = types
   .model('FeedbackStore', {
     isActive: types.optional(types.boolean, false),
-    rules: types.map(types.frozen({}))
+    rules: types.map(types.frozen({})),
+    showModal: types.optional(types.boolean, false),
+    allowSubjectAdvance: types.optional(types.boolean, false)
   })
+  .views(self => ({
+    get hideSubjectViewer () {
+      return flatten(Array.from(self.rules.values()))
+        .some(rule => rule.hideSubjectViewer)
+    },
+    get messages () {
+      return flatten(Array.from(self.rules.values()))
+        .map(rule => {
+          if (rule.success && rule.successEnabled) {
+            return rule.successMessage
+          } else if (!rule.success && rule.failureEnabled) {
+            return rule.failureMessage
+          }
+        })
```

Comment:
This should either return a default value for message, or chain the map with `.filter(Boolean)` to remove undefined messages from the array.

Reply:
Chain and filter sounds good, I think how Roger had initially, I'll refactor accordingly.

Reply:
I can't see those changes in the code.

Reply:
Ah sorry about that! I discarded a bunch of edits related to attempted middleware refactoring and these changes got discarded as well.

Reply:
Ok now should be ok.